src/system.rs

```diff
@@ -1071,6 +1071,101 @@ macro_rules! system {
             }
         }
 
+        /// Contains formatting tools for displaying quantities in this system.
+        pub mod fmt {
+
+            use $crate::lib::fmt;
+            use super::{Dimension, Quantity, Unit, Units, from_base};
+            use $crate::num::Num;
+            use $crate::Conversion;
+
+            /// Specifies the display style to use.
+            #[derive(Clone, Copy, Debug)]
+            pub enum DisplayStyle {
+                /// Display the value and a unit abbreviation, e.g. "1.0 m", "327 s".
+                Abbreviation,
+                /// Display the value and full unit name (pluralized as appropriate), e.g. "1 kilogram", "756 feet".
+                Description,
+            }
+
+            /// Provides an interface to specify a display style and unit.
+            ///
+            /// # Usage
+            /// ```
+            /// # use uom::si::Unit;
+            /// # use uom::si::{f32::*, length::{centimeter, meter}, fmt::{Arguments, DisplayStyle::*}};
+            /// let length = Length::new::<meter>(1.0);
+            /// let a = Arguments::new(centimeter, Description);
+            /// assert_eq!("100 centimeters", format!("{}", a.with(length)));
+            /// ```
+            #[allow(missing_debug_implementations)] // Prevent accidental direct use.
+            #[derive(Clone)]
+            pub struct Arguments<T: Unit> {
+                subunit: T,
+                style: DisplayStyle,
+            }
+
+            impl<T: Unit> Arguments<T> {
+
+                /// Creates a new `Argument` for displaying a quantity in the specified units and style.
+                pub fn new(subunit: T, style: DisplayStyle) -> Self {
+                    Self { subunit, style }
+                }
+
+                /// Specifies a quantity to display.
+                pub fn with<D, U, V>(&self, quantity: Quantity<D, U, V>) -> QuantityArguments<T, D, U, V>
```

Comment:
If `Arguments` was `Copy` I think this would be changed to taking self by value and not calling `clone()` down below.